### PR TITLE
SelfService V2

### PR DESCRIPTION
**Pull request by jamesmorgan**
* [x] Fix bug for invocation checks
* [x] Introduce configurable `minPricePerEdition`
* [x] Lower editions in 24 hours to 1
* [x] Introduce single access control contract so we can reuse this again
* [x] New Migration Scripts
	* [x] 132
	* [x] Create script to populate access controls
* [x] Deploy to Networks
	* [x] Ropsten
	* [x] Mainnet
* [x] Update `koda-tools`
	* [x] Ropsten
	* [x] Mainnet
* [x] Update `API`
	* [x] Cronjobs
		* [x] Ropsten
		* [x] Mainnet
* [x] Update Frontend
* [x] Ready for Review

### CODE REVIEW

File: contracts/v2/tools/SelfServiceEditionCurationV2.sol [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+      } else {
+        // if we have passed the time period simple clear out the fields and start the period again
+        invocationsInTimePeriod[msg.sender] = 1;
+        timeOfFirstInvocationInPeriod[msg.sender] = block.timestamp;
+      }
+
+    } else {
+      // initial the counters if not used before
+      invocationsInTimePeriod[msg.sender] = 1;
+      timeOfFirstInvocationInPeriod[msg.sender] = block.timestamp;
```
Comment (jamesmorgan) [inline contracts/v2/tools/SelfServiceEditionCurationV2.sol]: @andygray I was using `block.number` here so it never entered the timed frame checker

Comment (andygray) [reply contracts/v2/tools/SelfServiceEditionCurationV2.sol]: 👍

File: contracts/v2/tools/SelfServiceEditionCurationV2.sol [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+  mapping(address => bool) public allowedArtists;
+
+  // Config which enforces editions to not be over this size
+  uint256 public maxEditionSize = 100;
+
+  // When true this will skip the invocation in time period check
+  bool public disableInvocationCheck = false;
+
+  // Max number of editions to be created in the time period
+  uint256 public maxInvocations = 2;
```
Comment (andygray) [inline contracts/v2/tools/SelfServiceEditionCurationV2.sol]: I personally would have this at `1`

File: contracts/v2/tools/SelfServiceEditionCurationV2.sol [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+    kodaV2 = _kodaV2;
+    auction = _auction;
+  }
+
+  /**
+   * @dev Called by artists, create new edition on the KODA platform
+   */
+  function createEdition(
+    uint256 _totalAvailable,
+    uint256 _priceInWei,
```
Comment (andygray) [inline contracts/v2/tools/SelfServiceEditionCurationV2.sol]: Should we think of reducing `totalAvailable` to something lower than 100?

Comment (andygray) [reply contracts/v2/tools/SelfServiceEditionCurationV2.sol]: Should be `priceInWei` be at least a $1

Comment (jamesmorgan) [reply contracts/v2/tools/SelfServiceEditionCurationV2.sol]: I have added a configurable property so we can change this in the future 👍

File: contracts/v2/tools/SelfServiceEditionCurationV2.sol (hunk 2 above)
Comment (jamesmorgan) [reply contracts/v2/tools/SelfServiceEditionCurationV2.sol]: changed 👍

**Pull request merged**